src/highdicom/pr/sop.py

```diff
@@ -0,0 +1,1459 @@
+            modality_lut=modality_lut,
+            copy_modality_lut=copy_modality_lut,
+            softcopy_voi_luts=softcopy_voi_luts,
+            copy_voi_lut=copy_voi_lut,
+            icc_profile=icc_profile,
+            transfer_syntax_uid=transfer_syntax_uid,
+            **kwargs
+        )
+
+        # Palette Color Lookup Table
+        self._add_palette_color_lookup_table(palette_color_lut)
+
+    def _add_palette_color_lookup_table(
+        self,
+        palette_color_lut: PaletteColorLookupTable
+    ) -> None:
+        """Add attributes from the Palette Color Lookup Table module.
+
+        Parameters
+        ----------
+        palette_color_lut: highdicom.content.PaletteColorLookupTable
+            A palette color lookup table to apply to the image.
+
+        """
+        if not isinstance(palette_color_lut, PaletteColorLookupTable):
+            raise TypeError(
+                'Argument "palette_color_lut" must be of type '
+                'PaletteColorLookupTable.'
+            )
+        colors = ['Red', 'Green', 'Blue']
+
+        for color in colors:
+            desc_kw = f'{color}PaletteColorLookupTableDescriptor'
+            data_kw = f'{color}PaletteColorLookupTableData'
+            desc = getattr(palette_color_lut, desc_kw)
+            lut = getattr(palette_color_lut, data_kw)
+
+            setattr(self, desc_kw, desc)
+            setattr(self, data_kw, lut)
+
+        if hasattr(palette_color_lut, 'PaletteColorLookupTableUID'):
+            uid = palette_color_lut.PaletteColorLookupTableUID
+            self.PaletteColorLookupTableUID = uid
+
+
+class ColorSoftcopyPresentationState(_SoftcopyPresentationState):
+
+    """SOP class for a Color Softcopy Presentation State object.
+
+    A Color Softcopy Presentation State object includes instructions for
+    the presentation of a color image by software.
+
+    """
+
+    def __init__(
+        self,
+        referenced_images: Sequence[Dataset],
+        series_instance_uid: str,
+        series_number: int,
+        sop_instance_uid: str,
+        instance_number: int,
+        manufacturer: str,
+        manufacturer_model_name: str,
+        software_versions: Union[str, Tuple[str]],
+        device_serial_number: str,
+        content_label: str,
+        content_description: Optional[str] = None,
+        graphic_annotations: Optional[Sequence[GraphicAnnotation]] = None,
+        graphic_layers: Optional[Sequence[GraphicLayer]] = None,
+        graphic_groups: Optional[Sequence[GraphicGroup]] = None,
+        concept_name_code: Union[Code, CodedConcept, None] = None,
+        institution_name: Optional[str] = None,
+        institutional_department_name: Optional[str] = None,
+        content_creator_name: Optional[Union[str, PersonName]] = None,
+        content_creator_identification: Optional[
+            ContentCreatorIdentificationCodeSequence
+        ] = None,
+        icc_profile: Optional[ImageCmsProfile] = None,
```

Comment:
First, let's use `bytes` rather than the Pillow type. Second, the ICC Profile is required and in case of a true color image, we shouldn't assume any default color space (see related comment on `PseudoColorSoftcopyPresentationState`):

```suggestion
        icc_profile: bytes
```

Reply:
See also comment on `_SoftcopyPresentationState._add_icc_profile()`

Reply:
I'm quite concerned that this puts a large barrier in the way of the user. I don't understand this stuff really and if I had to manually specify an IMS profile just to create a presentation state, I wouldn't have a clue where to start. The average highdicom user who just wants to put a box over an image at a point is just going to give up pretty quickly.

Why is it so unreasonable to assume RGB?

Would a better solution be to copy this from the source image by default? Is it reasonable to assume it has one? I would guess many would not

Reply:
Moved the public API to bytes in e6ba4fa

Reply:
Color management is certainly a tricky subject. However, that's also why I think it's important that the caller has to explicitly provide a profile without us making too many assumptions about the color space.

> Why is it so unreasonable to assume RGB?

The problem is that RGB is not well defined. One can probably assume sRGB in most cases, but I would like the caller to make that assumption.

Reply:
@CPBridge as discussed, let's keep the parameter optional. If `None`, we should try to copy the ICC Profile from the referenced image. If the image does not contain an ICC Profile, we should either raise an exception or include a default sRGB profile. Currently, the standard says that an ICC Profile is required (module is mandatory and attribute type 1). However, given @dclunie's [comment](https://github.com/herrmannlab/highdicom/pull/139#discussion_r841255549), it may be that it may be fine to not include a profile if the IOD of the referenced image doesn't require one.

I'll leave the comment unresolved until we have further clarity on that.

Reply:
That's not what I said ... an ICC profile is always required in a color presentation state regardless of whether there was one in the referenced image or not - its absence would cause compliance checks (like in dciodvfy) to report an error.

Reply:
Thanks for the clarification @dclunie!